data/modules/Assassination/Assassination.lua

```diff
@@ -170,7 +171,16 @@ local makeAdvert = function (station)
 	local time = Engine.rand:Number(1, 4)
 	local due = Game.time + dist / max_ass_dist * time * 22*60*60*24 + Engine.rand:Number(7*60*60*24, 31*60*60*24)
 	local danger = Engine.rand:Integer(1,4)
-	local reward = Engine.rand:Number(2100, 7000) * danger
+
+	local tradeCargo = Game.player.totalCargo*0.81 --the rest 0.19 is used for fuel.
+	local rewardRatio = 3/4-- For getting that ratio of average effective trading profit
+
+	--- Fighters should have independed from cargo reward. For that we increase minimum reward depending on number of missiles that player can acquire.
+	local missiles = Game.player:GetEquipSlotCapacity("missile")
+	local minimumRewardRatio = math.max(rewardRatio, Calibration.Fighting(missiles))
+
+	local profit = math.max(Calibration.profit(tradeCargo) * rewardRatio, Calibration.minProfit * minimumRewardRatio)  
+	local reward = Engine.rand:Number(2100/8259 * profit, 7000/8259 * profit) * danger --8259 is the current estimated profit from assassination module.
```

Comment:
IMO the reward should be based on the service rendered, not the capabilities of whoever gets the job done. But even if we'd take the player's ship into account, why do we care about the cargo space?

Reply:
Might be done in a different way? I mean a good provider could indeed demand more for their services (amateur logo designer vs professional with decades of experience for example would ask for quite different money.
Like negotiations. I know that this would be quite out-of-scope here though, but if the player could ask for more money because their costs would be higher on that mission, and the mission provider would either accept or refuse based on a mission variable and player reputation for example

Reply:
Figure 6 of my article 
![current_state](https://user-images.githubusercontent.com/5658995/134398969-394acc9a-da95-414e-9047-8eafc52cb3df.png)
It is clear that in current state missions are auxiliary till player starts to trade effectively. One can see that he does not need much more from 83t free cargo to get the game's maximum profit. In this PR, since initial trading profits are reduced, missions are more important but still only in the start of game life. At middle and farther of economy game they become insignificant. It is needed a clear global direction. What is the role of missions' reward compared to trading profit? I kept most important missions' reward close to trading profit every time, so as not to loose their significance. If player can not find something useful to trade _(he relentlessly seek for his best commodity to trade in this PR)_, he may choose a mission without losing much. 

175-176 lines, along with first part of max in line 182, define rewards as lightly auxiliary to trading profit, to 3/4 of it. But fighters have small free cargo and they fail as  traders _(Calibration.profit(tradeCargo) becomes extremely small)_. The profit function sees them as merchants. So, we get the maximum of their  profit as merchant and as fighters by the Calibration.Fighting(missiles) function. Many ships, in their json files, had both "mercenary", "merchant" roles and we need both their mercenary and merchant profits to compare.